traitlets/traitlets.py

```diff
@@ -1204,10 +1204,28 @@ class Int(TraitType):
     default_value = 0
     info_text = 'an int'
 
+    def __init__(self, default_value=NoDefaultSpecified,
+                 allow_none=None, **kwargs):
+        self.min = kwargs.pop('min', None)
+        self.max = kwargs.pop('max', None)
+        super(Int, self).__init__(default_value=default_value,
+                                  allow_none=allow_none, **kwargs)
+
     def validate(self, obj, value):
-        if isinstance(value, int):
-            return value
-        self.error(obj, value)
+        if not isinstance(value, int):
+            self.error(obj, value)
+        if self.max is not None and value > self.max:        
+            raise TraitError("The value of the '%s' trait of %s instance should "
+                             "not be greater than %s, but a value of %s was "
+                             "specified" % (self.name, class_of(obj),
+                                            self.max, repr_type(value)))
```

Comment:
`repr_type()` will return something like `200 <class 'int'>`. Given that at this point it must be the correct type, we can just show its repr with `%r` in the format string.